src/app/map/[id]/components/Map.tsx

```diff
@@ -60,17 +71,48 @@ export default function Map({
     ready,
     setReady,
   } = useContext(MapContext);
-  const { viewConfig } = useMapViews();
+  const { viewConfig, view } = useMapViews();
   const { mapConfig } = useMapConfig();
+  const { getDataSourceById } = useDataSources();
+
+  // Tag resending functionality
+  const trpc = useTRPC();
+  const { mutate: tagRecords } = useMutation(
+    trpc.mapView.tagRecordsWithViewName.mutationOptions({
+      onSuccess: () => {
+        toast.success("Tagging records in the background");
+      },
+      onError: () => {
+        toast.error("Failed to tag records with view name");
+      },
+    }),
+  );
   const {
     deleteTurf,
     insertTurf,
     updateTurf,
-    turfs,
     searchMarker,
     placedMarkers,
     markerQueries,
   } = useContext(MarkerAndTurfContext);
+
+  // Get the first data source and its view for tag preview
+  const firstDataSourceId = getDataSourceIds(mapConfig)[0];
```

Comment:
note: this doesn't seem right. When the records are tagged from the table view, the data source that is open is the one that is tagged. In this component, it will always tag the first data source - which may not be what the user wants.